### PR TITLE
Proposed fix for #768

### PR DESCRIPTION
**Pull request by sanderboer**
This is a duplicate of a previous PR, I am deleting the first one. My v0.6.0 branch will a merge of a number of fixes and this pr is cherrypicked.

### CODE REVIEW

**Comment by aothms** [comment]
Thanks I need to understand this a bit better. This is a recursive function so making this static or thread_local means these curve handles are shared between recursion levels (?) That is counter-intuitive to me.

**Comment by sanderboer** [comment]
Maybe it is better to do this ?
```
thread_local Handle(Geom_Curve) c0;
c0 = BRep_Tool::Curve(wd->Edge(i + 1), u11, u12);
thread_local Handle(Geom_Curve) c1;
c1 = BRep_Tool::Curve(wd->Edge(j + 1), u21, u22);
```
I am unclear if c++ allocates new memory if you initialize a static (or thread_local) variable with the result of a function. I would think the code above is equivalent with the patch, but it's still c++.

**Comment by sanderboer** [comment]
Wait, I didn't read your comment right, I was worried about memory leaking. 
It is possible that the last value of c0 and c1 carry over to the next recursion level. But that value is clobbered with the new value of BRep_Tool::Curve.

File: src/ifcgeom/IfcGeomFunctions.cpp [hunk 1]
```diff
@@ -3691,6 +3691,8 @@ bool IfcGeom::Kernel::wire_intersections(const TopoDS_Wire& wire, TopTools_ListO
 	}
 	
 	bool intersected = false;
+        thread_local Handle(Geom_Curve) c0;
+        thread_local Handle(Geom_Curve) c1;
```
Comment (sanderboer) [inline src/ifcgeom/IfcGeomFunctions.cpp]: Turns out, this is a not smart move. This reintroduces the segfault.

**Comment by aothms** [comment]
This was addressed by the check on `IsParallel`. But many thanks for the investigations.

**Pull request closed without merging**